internal/service/cloudbackupsnapshotexportjob/resource_cloud_backup_snapshot_export_job_test.go

```diff
@@ -10,43 +10,65 @@ import (
 	"github.com/hashicorp/terraform-plugin-testing/terraform"
 	"github.com/mongodb/terraform-provider-mongodbatlas/internal/common/conversion"
 	"github.com/mongodb/terraform-provider-mongodbatlas/internal/testutil/acc"
-	matlas "go.mongodb.org/atlas/mongodbatlas"
 )
 
-func TestAccBackupRSBackupSnapshotExportJob_basic(t *testing.T) {
-	acc.SkipTestForCI(t)
+var (
+	resourceName         = "mongodbatlas_cloud_backup_snapshot_export_job.test"
+	dataSourceName       = "data.mongodbatlas_cloud_backup_snapshot_export_job.test"
+	dataSourcePluralName = "data.mongodbatlas_cloud_backup_snapshot_export_jobs.test"
+)
+
+func TestAccBackupSnapshotExportJob_basic(t *testing.T) {
+	resource.ParallelTest(t, *basicTestCase(t))
+}
+
+func basicTestCase(tb testing.TB) *resource.TestCase {
+	tb.Helper()
+	acc.SkipTestForCI(tb) // needs AWS IAM role and S3 bucket
+
 	var (
-		snapshotExportJob matlas.CloudProviderSnapshotExportJob
-		resourceName      = "mongodbatlas_cloud_backup_snapshot_export_job.test"
-		projectID         = os.Getenv("MONGODB_ATLAS_PROJECT_ID")
-		bucketName        = os.Getenv("AWS_S3_BUCKET")
-		iamRoleID         = os.Getenv("IAM_ROLE_ID")
+		projectID  = os.Getenv("MONGODB_ATLAS_PROJECT_ID")
+		bucketName = os.Getenv("AWS_S3_BUCKET")
+		iamRoleID  = os.Getenv("IAM_ROLE_ID")
 	)
-	resource.ParallelTest(t, resource.TestCase{
-		PreCheck:                 func() { acc.PreCheck(t); acc.PreCheckS3Bucket(t) },
+
+	return &resource.TestCase{
+		PreCheck:                 func() { acc.PreCheck(tb); acc.PreCheckS3Bucket(tb) },
 		ProtoV6ProviderFactories: acc.TestAccProviderV6Factories,
-		CheckDestroy:             testAccCheckMongoDBAtlasBackupSnapshotExportJobDestroy,
+		CheckDestroy:             checkDestroy,
 		Steps: []resource.TestStep{
 			{
-				Config: testAccMongoDBAtlasBackupSnapshotExportJobConfig(projectID, bucketName, iamRoleID),
+				Config: configBasic(projectID, bucketName, iamRoleID),
 				Check: resource.ComposeTestCheckFunc(
-					testAccCheckMongoDBAtlasBackupSnapshotExportJobExists(resourceName, &snapshotExportJob),
+					checkExists(resourceName),
 					resource.TestCheckResourceAttr(resourceName, "project_id", projectID),
 					resource.TestCheckResourceAttr(resourceName, "bucket_name", "example-bucket"),
 					resource.TestCheckResourceAttr(resourceName, "cloud_provider", "AWS"),
+					resource.TestCheckResourceAttrSet(resourceName, "iam_role_id"),
+
+					resource.TestCheckResourceAttr(dataSourceName, "project_id", projectID),
+					resource.TestCheckResourceAttr(dataSourceName, "bucket_name", "example-bucket"),
```

Comment:
applies to 3 checks in this test and other 3 in other resource test
```suggestion
					resource.TestCheckResourceAttr(dataSourceName, "bucket_name", bucketName),
```